### PR TITLE
Implementation of the REST API

### PR DESCRIPTION
**Pull request by Oleh-Stefanovych**
The most basic changes:
- Rewriting existing requests with REST API;
- Implementation of new requests for files;
- Expanding file model;
- Implementation of chaining for router;
- Providing `Application` class.

The ESLint result:
image

### CODE REVIEW

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, PTAL

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, PTAL.

File: app/model/file/index.js [hunk 1]
```diff
@@ -1,31 +1,87 @@
...
+   * @return {string} - the type of the file.
+   */
+  get type() {
+    return this._type;
+  }
+
+  /**
+   * Returns a JSON file representation.
+   *
+   * @return {string} - a JSON file representation.
```
Comment (dkalinin) [inline app/model/file/index.js]: It returns object.

File: app/presenter/login-presenter/index.js [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
    *
-   * @param {Element} root - a parent {@link Element} to append the {@link LoginPage}.
-   * @param {Router} router - a router for change the address of the web page.
+   * @param {Element} root - a parent {@link Element} to append the page.
    * @param {Store} store - a global storage of application states.
    */
-  constructor(root, router, store) {
-    super(root, router, LoginPage, store);
-    this.page.onLoginSubmit((loginData) => this._loginSubmittedHandler(loginData));
+  constructor(root, store) {
```
Comment (dkalinin) [inline app/presenter/login-presenter/index.js]: Login and Registration presenter needs services and router only.

File: app/presenter/page-404-presenter/index.js [hunk 3]
```diff
@@ -2,20 +2,20 @@ import {Presenter} from '../';
...
    * and the processing of its events.
    *
-   * @param {Element} root - a parent {@link Element} to append the {@link Page404}.
-   * @param {Router} router - a router for change the address of the web page.
+   * @param {Element} root - a parent {@link Element} to append the page.
    * @param {Store} store - a global storage of application states.
    */
-  constructor(root, router, store) {
-    super(root, router, Page404, store);
+  constructor(root, store) {
```
Comment (dkalinin) [inline app/presenter/page-404-presenter/index.js]: It needs nothing.

File: app/service/request-service/index.js [hunk 4]
```diff
@@ -2,25 +2,25 @@
...
+  // TODO:2019-08-22:Oleh.Stefanovych: Implementation of a sending request is required.
   /**
    * Sends the POST HTTP request to the server.
    *
    * @param {string} url - an URL to which the request will be sent.
    * @param {string} json - a body of the request to be sent.
    * @return {Promise<any>} - a server response.
    */
-  sendPost(url, json) {
+  sendPost(url, json = '') {
```
Comment (dkalinin) [inline app/service/request-service/index.js]: Can be a file.

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, fixed, PTAL.

File: app/index.js [hunk 5]
```diff
@@ -4,26 +4,32 @@ import {Router} from './router';
...
+import {RegistrationRequestService} from './service/registration-request-service';
+import {FileRequestService} from './service/file-request-service';
 
+const APPLICATION_URL = 'https://app.com';
 const root = document.getElementById('index');
 
-const initialState = {};
-
-const store = new Store(initialState);
+const requestService = new RequestService();
```
Comment (dkalinin) [inline app/index.js]: Pass App_URL here.

File: app/store/actions.js [hunk 6]
```diff
@@ -1,23 +1,34 @@
...
-  removeFile(store, file) {
-    store.commit('setFileListStatus', `Removing file '${file.name}'...`);
+  getRootFolder(store) {
+    return () => {
+      store.commit('setFileListStatus', 'Connecting to drive...');
+      store.requestServices.fileRequestService.getRootFolder()
+          .then((response) => {
+            store.commit('setCurrentFolder', JSON.parse(response)['root-folder']);
+            store.dispatch('fetchFiles', store.state.currentFolder);
+          });
```
Comment (dkalinin) [inline app/store/actions.js]: Please handle catch or finally to clear file list status.

File: app/store/actions.js [hunk 7]
```diff
@@ -1,23 +1,34 @@
...
-        .then(() => {
-          store.dispatch('fetchFiles');
-        });
+  removeFile(store) {
+    return (file) => {
+      store.commit('setFileListStatus', `Removing file '${file.name}'...`);
+      store.requestServices.fileRequestService.removeFile(file.id)
+          .then(() => {
+            store.dispatch('fetchFiles', store.state.currentFolder);
+          });
```
Comment (dkalinin) [inline app/store/actions.js]: Please handle catch or finally to clear file list status.

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, fixed, PTAL.

**Pull request merged**